src/harness/sas_test_harness.py

```diff
@@ -284,71 +284,74 @@ def generateCbsdReferenceId(fcc_id, serial_number):
-    # be incorrect or missing.
+    # Assign grant expiry time to grant objects.
     grant_time_stamp = datetime.now().today() + timedelta(hours=24)
-    grant_expiry_time = time.strftime(grant_time_stamp.replace(microsecond=0).isoformat()) + 'Z'
+    grant_expire_time = time.strftime(grant_time_stamp.replace(microsecond=0).isoformat()) + 'Z'
 
     internal_data = {
         'id': 'cbsd/{}'.format(encoded_cbsd_id),
         'registration': {
-            'fccId': cbsd_record['fccId'],
-            'cbsdCategory': cbsd_record['cbsdCategory'],
-            'callSign': cbsd_record['callSign'],
-            'userId': cbsd_record['userId'],
-            'airInterface': cbsd_record['airInterface'],
-            'measCapability': cbsd_record['measCapability'],
-            'installationParam': cbsd_record['installationParam']
+            'fccId': registration_request['fccId'],
+            'cbsdCategory': registration_request['cbsdCategory'],
+            'callSign': registration_request['callSign'],
+            'airInterface': registration_request['airInterface'],
+            'measCapability': registration_request['measCapability'],
+            'installationParam': registration_request['installationParam']
         }
     }
 
     # Check if groupingParam optional parameter present in CBSDRecord then
     # add to registration object.
-    if cbsd_record.has_key('groupingParam'):
-      internal_data['registration'].update({'groupingParam': cbsd_record['groupingParam']})
-    internal_data_grant = {}
-    internal_data_grant['grants'] = []
+    if registration_request.has_key('groupingParam'):
+      internal_data['registration'].update({'groupingParam': registration_request['groupingParam']})
+
+    internal_data['grants'] = []
 
     # Read all grant records if mulitiple grant records are associated with single cbsd.
-    for grant_record in grant_records:
+    for grant_request in grant_requests:
       grant_data = {}
-      if not grant_record.has_key('id'):
-        grant_data['id'] = 'SAMPLE_ID_{:05}'.format(random.randrange(1, 10 ** 5))
-      else:
-        grant_data['id'] = grant_record['id']
-      grant_data['operationParam'] = grant_record['operationParam']
-      if not grant_record.has_key('channelType'):
-        grant_data['channelType'] = 'GAA'
-      else:
-        grant_data['channelType'] = grant_record['channelType']
-      if not grant_record.has_key('grantExpireTime'):
-        grant_data['grantExpireTime'] = grant_expiry_time
+      if grant_request.has_key('id'):
+        grant_data['id'] = grant_request['id']
+      if grant_request.has_key('operationParam'):
+        grant_data['operationParam'] = grant_request['operationParam']
+      if grant_request.has_key('channelType'):
+        grant_data['channelType'] = grant_request['channelType']
+
+      # grantExpireTime is mandatory in GrantData Object.
+      if not grant_request.has_key('grantExpireTime'):
```

Comment:
> kate-harrison 19 hours ago  Member
> If you are taking GrantRequest as input, it will never have the field "grantExpireTime". Please carefully decide what your input should be and then work from there.
> 
>   @jni2000
> jni2000 16 hours ago  Member
> As per SAS-SAS TS grantObject requires to update with grantExpireTime value


I agree that it is mandatory in the GrantData object. My point is that it will _never_ be in the GrantRequest object so you always have to fill it manually.

Reply:
grant_expire_time is assigned directly